rfd/000?-database-access.md

```diff
@@ -0,0 +1,512 @@
+dispatch to an appropriate service.
+
+## Authentication
+
+In this model, there are 3 points where authentication needs to happen.
+
+### Database client <---> proxy
+
+**Mutual TLS.** Database clients, such as psql, will use short-lived x509
+certificates issued by Teleport in order to authenticate with the proxy.
+
+The certificate metadata includes (via extensions) routing information about
+the target database which users log into using `tsh db login` command (see
+below for UX).
+
+### Proxy <---> database service
+
+**Mutual TLS.** The connection that is passed from proxy to a database service
+is upgraded to TLS as well in order to be able to pass identity information
+over the reverse tunnel.
+
+In order to do that, proxy will generate an ephemeral certificate signed
+by the user authority and re-encode all identity and routing information
+extracted from the certificate supplied by the client into it.
+
+### Database service <---> database instance
+
+**Mutual TLS (onprem).**  When connecting to an onprem database instance, the
+database service will use a client certificate for authentication, which it
+will generate on the fly according to the database requirements (for example,
+PostgreSQL requires the database user name to be encoded in the certificate
+as common name).
+
+This also means that the database needs to be configured with Teleport's
+certificate authority and (in general case) use server certificate issued by
+Teleport which can be done using `tctl auth sign` command (see below for UX).
+
+**Password auth (RDS/Aurora).** Amazon RDS/Aurora don't support client cert
+authentication. Instead, they support IAM authentication so Teleport database
+service will generate a short-lived authentication token using RDS API that
+will be used as a password:
+
+https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.IAMDBAuth.html
+
+In order to be able to verify RDS/Aurora server certificate, users will need
+to download RDS root certificate and configure the database service to trust
+it:
+
+https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.SSL.html
+
+## Configuration
+
+### Teleleport database service
+
+The following new configuration section is added to the Teleport file config:
+
+```yaml
+# New global key housing the database service configuration.
+db_service:
+  # Enable or disable the database service.
+  enabled: "yes"
+  # List of the database this service is proxying.
+  databases:
+    # Database instance name, used to refer to an instance in CLI like tsh.
+  - name: "postgres-prod"
+    # Optional free-form verbose description of a database instance.
+    description: "Production instance of PostgreSQL 13.0"
+    # Database procotol, only "postgres" is supported initially.
+    protocol: "postgres"
+    # Database connection endpoint, should be reachable from the service.
+    endpoint: "postgres.internal.example.com:5432"
```

Comment:
this should be consistent with application access vocabulary, for example it uses `uri` instead of `endpoint`. Please cross reference for similar concepts and use similar terms here and below

Reply:
That's how I named it originally actually but for some reason thought that endpoint would fit better for databases. I'll rename it back to uri.